### PR TITLE
Rhel dev enable repo from hd

### PR DESCRIPTION
**Pull request by jkonecny12**
Enable installation from an installation tree extracted on HDD.

*Resolves: rhbz#1502323*

### CODE REVIEW

File: pyanaconda/image.py [hunk 1]
```diff
@@ -116,6 +116,20 @@ def findFirstIsoImage(path):
     return None
 
 
+def verify_valid_installtree(path):
+    """Check if the given path is a valid installtree repository
+
```
Comment (M4rtinK) [inline pyanaconda/image.py]: Maybe add:

`:param str path: install three path`

File: pyanaconda/image.py [hunk 2]
```diff
@@ -116,6 +116,20 @@ def findFirstIsoImage(path):
...
+    :returns: True if repository is valid false otherwise
```
Comment (M4rtinK) [inline pyanaconda/image.py]: I suggest:

```
:returns: True if repository is valid False otherwise
:rtype: bool
```

File: pyanaconda/image.py [hunk 3]
```diff
@@ -116,6 +116,20 @@ def findFirstIsoImage(path):
...
+    """
+    # TODO: This can be enhanced for check for repodata folder.
```
Comment (M4rtinK) [inline pyanaconda/image.py]: *to* check for repodata folder ?

File: pyanaconda/payload/__init__.py [hunk 4]
```diff
@@ -1186,13 +1196,21 @@ def _find_and_mount_iso(self, device, device_mount_dir, iso_path, iso_mount_dir)
...
 
         return iso_path
 
+    def _setup_install_tree(self, device, install_tree_path, device_mount_dir):
+        self._setupDevice(device, mountpoint=device_mount_dir)
+        path = os.path.normpath("%s/%s" % (device_mount_dir, install_tree_path))
+
+        if not verify_valid_installtree(path):
+            device.teardown(recursive=True)
+            raise PayloadSetupError("failed to find valid installation tree")
```
Comment (poncovka) [inline pyanaconda/payload/__init__.py]: Make a real sentence from the string, please.

File: pyanaconda/ui/gui/spokes/installation_source.py [hunk 5]
```diff
@@ -658,13 +659,26 @@ def status(self):
...
         else:
             return _("Nothing selected")
 
+    def _get_harddrive_partition_name(self):
+        devices = device_matches(self.data.method.partition)
+        if not devices:
+            log.warning("Device for method harddrive partition can't be found!")
+            return ""
+        elif len(devices) > 1:
+            log.warning("More than one device is found for method harddrive partition!")
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/installation_source.py]: I am not sure if the term "method harddrive partition" makes sense.

File: pyanaconda/ui/gui/spokes/installation_source.py [hunk 6]
```diff
@@ -863,6 +875,17 @@ def _initialize(self):
         # report that the source spoke has been initialized
         self.initialize_done()
 
+    def _enable_autodetect_with_device(self, device):
+        device_label = getattr(device.format, "label", "") or ""
+        self._enable_autodetect(device.name, device_label)
+
+    def _enable_autodetect(self, device_name, device_label):
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/installation_source.py]: This method is more about showing autodetected devices than about enabling the autodetection. Could you change the name, please?

File: pyanaconda/ui/gui/spokes/installation_source.py [hunk 7]
```diff
@@ -863,6 +875,17 @@ def _initialize(self):
         # report that the source spoke has been initialized
         self.initialize_done()
 
+    def _enable_autodetect_with_device(self, device):
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/installation_source.py]: Same here.

File: pyanaconda/ui/gui/spokes/installation_source.py (hunk 5 above)
Comment (jkonecny12) [reply pyanaconda/ui/gui/spokes/installation_source.py]: Did `harddrive method partition` sounds better?

**Comment by jkonecny12** [comment]
**Updated:**
* Most of the items above. Thanks for suggestions!

File: pyanaconda/ui/gui/spokes/installation_source.py (hunk 5 above)
Comment (VladimirSlavik) [reply pyanaconda/ui/gui/spokes/installation_source.py]: If there's internally something called `harddrive partition`, how about quotes or such?

`log.warning("Device for 'harddrive partition' method can't be found!")`

Comment (poncovka) [reply pyanaconda/ui/gui/spokes/installation_source.py]: What about `Device for installation from HDD can't be found!`? I guess I don't like the `method` there. It is an internal name.

Comment (jkonecny12) [reply pyanaconda/ui/gui/spokes/installation_source.py]: Thanks @VladimirSlavik and @poncovka for a great suggestions! Fixed.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Warning message changed.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fix tests by adding python3-coverage as test dependency.

**Pull request merged**